game-headed/src/main/java/org/triplea/game/client/ui/javafx/screens/MapSelection.java

```diff
@@ -0,0 +1,149 @@
+import java.util.stream.Collectors;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContentDisplay;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.TilePane;
+import javafx.scene.web.WebView;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.triplea.game.client.parser.DefaultGameDetector;
+import org.triplea.game.client.parser.GameDetector;
+import org.triplea.game.client.ui.javafx.screen.ControlledScreen;
+import org.triplea.game.client.ui.javafx.screen.ScreenController;
+import org.triplea.game.client.ui.javafx.util.FxmlManager;
+import org.triplea.util.LocalizeHtml;
+
+public class MapSelection implements ControlledScreen<ScreenController<FxmlManager>> {
+
+  private final GameDetector gameDetector;
+  private final BiFunction<String, String, String> linkLocalizer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private Node root;
+
+  @FXML private TilePane mapContainer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private Node previewContainer;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  @FXML
+  private WebView previewWindow;
+
+  @FXML private Button selectButton;
+  @FXML private Button detailsButton;
+
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  private GameChooserEntry selectedGame;
+
+  private boolean loaded = false;
+
+  private ScreenController<FxmlManager> screenController;
+
+  // Constructor used via FXML reflection
+  // to initialize this controller.
+  @SuppressWarnings("unused")
+  public MapSelection() {
+    this(new DefaultGameDetector(), LocalizeHtml::localizeImgLinksInHtml);
+  }
+
+  @VisibleForTesting
+  MapSelection(
+      final GameDetector gameDetector, final BiFunction<String, String, String> linkLocalizer) {
+    this.gameDetector = gameDetector;
+    this.linkLocalizer = linkLocalizer;
+  }
+
+  private Node createMapListing(final GameChooserEntry gameChooserEntry) {
+    final var button = new Button(gameChooserEntry.getGameName());
+    // Placeholder image
+    final var imageView =
+        new ImageView(new Image("https://triplea-game.org/images/missing_map.png", true));
+    imageView.setPreserveRatio(true);
+    imageView.setFitHeight(150);
+    imageView.setFitWidth(170);
+    button.setGraphic(imageView);
+    button.setWrapText(true);
+    button.setPrefSize(200, 200);
+    button.setContentDisplay(ContentDisplay.TOP);
+    button.setOnAction(
+        event -> {
+          selectedGame = gameChooserEntry;
+          selectButton.setDisable(false);
+          detailsButton.setDisable(false);
+          mapContainer.getChildren().forEach(node -> node.getStyleClass().remove("selected"));
```

Comment:
If the structure of mapContainer is changed, will this break? What kind of children nodes do we expect here? Would it be safer to be more direct about the unsetting of 'selected' by assigning the nodes of mapContainer to a collection directly on construction, then we could be more explicit about which nodes are operating on.

Reply:
Well no, but maybe.
In the FXML file `mapContainer` is a `TilePane`, with a intermediate loading icon.
Once the screen is shown and we have calculated all the available maps asynchronously, we clear this container and add all the games as buttons with icons to this container.
So this particular container is completely managed by the java code in this class once the games are loaded, so nothing can really break from outside influence at this point, but this isn't necessarily the case for all containers that are managed dynamically.